projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php

```diff
@@ -386,6 +386,29 @@ protected function __construct() {
 		wp_register_style( 'grunion.css', GRUNION_PLUGIN_URL . 'css/grunion.css', array(), JETPACK__VERSION );
 		wp_style_add_data( 'grunion.css', 'rtl', 'replace' );
 
+		wp_enqueue_style(
+			'jquery-ui-css',
+			'https://code.jquery.com/ui/1.13.2/themes/base/jquery-ui.css',
+			array(),
+			JETPACK__VERSION
+		);
+
+		wp_enqueue_script(
+			'jquery-ui',
+			'https://code.jquery.com/ui/1.13.2/jquery-ui.js',
```

Comment:
Yeah... we might not want to do that. At the very least it should be coming from our CDN or maybe we want to copy that into Jetpack altogether. @Automattic/jetpack-crew can we get your opinion on this?

Reply:
Are the parts of jquery-ui actually needed here [already included in WordPress core](https://developer.wordpress.org/reference/functions/wp_enqueue_script/#default-scripts-and-js-libraries-included-and-registered-by-wordpress)? Or can something included in core (whether jquery or something in the Gutenberg `@wordpress/*` packages) be used instead? That would be the best option.

If bundling, avoid copying in the actual source files into the repo. Prefer bringing in the code as a package via pnpm, then use webpack to build an appropriate minified bundle.[^1] Keep an eye on the size of the bundles.

If we do need to include a non-bundled script, it would probably be best to prefer loading from an Automattic CDN. Compare pMz3w-g6E-p2 for example.

[^1]: Note there's probably a bunch of existing code violating this suggestion. Until recently using packages and bundles was constrained by the way the code was used on wpcom, but with the completion of de-Fusioning those constraints are probably now gone. PRs modernizing things would be appreciated!

Reply:
Thanks for the answer @anomiex!
So, yes! The `jquery-ui-selectmenu` is included in WordPress core.
Regarding the styles for the jQuery UI, I saw [this note](https://developer.wordpress.org/reference/functions/wp_enqueue_style/#notes) and, inspecting the [wp_default_styles](https://github.com/WordPress/wordpress-develop/blob/6.1/src/wp-includes/script-loader.php#L1470-L1757) code, I didn't find the jQuery UI styles we need.
Is the approach for CSS files different?
Can we load it by hitting `//code.jquery.com/ui/1.13.2/themes/base/jquery-ui.css` directly, or should we download the file and store it?
The latter seems to be the option for the [jquery-ui-datepicker](https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/jetpack/modules/contact-form/css/jquery-ui-datepicker.css) on Jetpack.

Reply:
>  should we download the file and store it?

That's what I would recommend, yep. We should be able to only bring in what we need, and serve it locally. Our existing system will allow to concatenate the styles into the existing admin styles.

Reply:
Got it! Thanks @jeherve!

Reply:
My WTF for today: Why doesn't WordPress ship any CSS for the jQuery UI stuff it bundles? In https://core.trac.wordpress.org/ticket/52022 someone says

> The intent is for plugins or themes that rely on jQuery UI to include the styles them selves, so that it matches their expected design or visuals. Any plugins or themes doing so would need to update their styles to match the updated version of jQuery UI.

which seems to me like a great recipe for plugins and themes to wind up stomping on each other's styles. But, since that's the official way to do it, :shrug: